### PR TITLE
Update trimmer failure code handling and statistics parsing

### PR DESCRIPTION
**Pull request by kenissur**
- Updated `read_and_parse_sorter_statistics_csv` to accept a metrics shortlist, and removed Failed_QC_reads by default because that is no longer how unmatched reads are handled.
- Modified `read_trimmer_failure_codes` to include an option to exclude failed RSQ reads.
- Added logic to conditionally exclude RSQ reads.
- Added a new unit tests to validate processing of recent trimmer failure code file.

### CODE REVIEW

File: src/ppmseq/tests/resources/412884-L6860-Z0293-CATGTGAGCGGTGAT_trimmer-failure_codes.csv [hunk 1]
```diff
@@ -0,0 +1,9 @@
+read group,code,format,segment,reason,failed read count,total read count
+Z0293,2,ppmSeq_v2,start,sequence was too long,1175579,1267433062
+Z0293,8,ppmSeq_v2,start,rsq file,169314209,1267433062
```
Comment (GatKrieger) [inline src/ppmseq/tests/resources/412884-L6860-Z0293-CATGTGAGCGGTGAT_trimmer-failure_codes.csv:3]: Does it make sense that 13% of reads failed rsq?

File: src/ppmseq/tests/unit/test_ppmSeq_utils.py [hunk 2]
```diff
@@ -105,6 +111,18 @@ def compare_records(records1, records2):
     assert not records_diff, "Differences found in records"
 
 
+def test_read_trimmer_failure_codes_ppmseq(tmpdir):
+    df_trimmer_failure_codes, df_metrics = read_trimmer_failure_codes_ppmseq(
+        trimmer_failure_codes_csv_ppmseq_v1_incl_failed_rsq
+    )
+    assert (
```
Comment (GatKrieger) [inline src/ppmseq/tests/unit/test_ppmSeq_utils.py:118]: maybe use variables to contain these numbers

File: src/ppmseq/ugbio_ppmseq/ppmSeq_utils.py [hunk 3]
```diff
@@ -902,7 +902,9 @@ def read_trimmer_failure_codes_ppmseq(trimmer_failure_codes_csv: str):
     AssertionError
         If the columns are not as expected
     """
-    df_trimmer_failure_codes = read_trimmer_failure_codes(trimmer_failure_codes_csv)
+    df_trimmer_failure_codes = read_trimmer_failure_codes(
+        trimmer_failure_codes_csv, include_failed_rsq=False, add_total=True
```
Comment (GatKrieger) [inline src/ppmseq/ugbio_ppmseq/ppmSeq_utils.py:906]: Just to understand: now rsq filtering is done by trimmer. you can either include the rsq filtering information in the trimmer failure code file or not. default is not to include it. So how can I know hoe many reads were filtered for rsq?

**Review by GatKrieger — CHANGES_REQUESTED** [review]

File: src/ppmseq/ugbio_ppmseq/ppmSeq_utils.py (hunk 3 above)
Comment (kenissur) [reply src/ppmseq/ugbio_ppmseq/ppmSeq_utils.py:906]: This is not a metric we ever tracked in the application QC

**Review by GatKrieger — APPROVED** [review]

**Pull request merged**